src/plugins/event-plugins/ResourcePlugin.ts

```diff
@@ -49,73 +52,67 @@ export class ResourcePlugin extends InternalPlugin {
+                // Always record
+                this.recordResourceEvent(entry);
+            } else if (
+                this.sampleCount < this.config.eventLimit &&
+                this.config.sampleTypes.includes(fileType)
+            ) {
+                // Only sample first N
+                this.recordResourceEvent(entry);
+                this.sampleCount++;
+            }
         }
     };
 
-    recordResourceEvent = ({
-        name,
-        startTime,
-        initiatorType,
-        duration,
-        transferSize
-    }: PerformanceResourceTiming): void => {
-        const pathRegex =
-            /.*\/application\/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\/events/;
-        const entryUrl = new URL(name);
-        if (
-            entryUrl.host === this.context.config.endpointUrl.host &&
-            pathRegex.test(entryUrl.pathname)
-        ) {
-            // Ignore calls to PutRumEvents (i.e., the CloudWatch RUM data
-            // plane), otherwise we end up in an infinite loop of recording
-            // PutRumEvents.
-            return;
-        }
-
-        if (this.context?.record) {
-            const eventData: ResourceEvent = {
-                version: '1.0.0',
-                initiatorType,
-                startTime,
-                duration,
-                fileType: getResourceFileType(name, initiatorType),
-                transferSize
-            };
-            if (this.context.config.recordResourceUrl) {
-                eventData.targetUrl = name;
-            }
-            this.context.record(PERFORMANCE_RESOURCE_EVENT_TYPE, eventData);
-        }
+    recordResourceEvent = (r: PerformanceResourceTimingPolyfill): void => {
+        this.context?.record(PERFORMANCE_RESOURCE_EVENT_TYPE, {
+            version: '2.0.0',
+            name: this.context.config.recordResourceUrl ? r.name : undefined,
+            entryType: RESOURCE,
+            startTime: r.startTime,
+            duration: r.duration,
+            connectStart: r.connectStart,
+            connectEnd: r.connectEnd,
+            decodedBodySize: r.decodedBodySize,
+            domainLookupEnd: r.domainLookupEnd,
+            domainLookupStart: r.domainLookupStart,
+            fetchStart: r.fetchStart,
+            encodedBodySize: r.encodedBodySize,
+            initiatorType: r.initiatorType,
+            nextHopProtocol: r.nextHopProtocol,
+            redirectEnd: r.redirectEnd,
+            redirectStart: r.redirectStart,
+            renderBlockingStatus: r.renderBlockingStatus,
+            requestStart: r.requestStart,
+            responseEnd: r.responseEnd,
+            responseStart: r.responseStart,
+            secureConnectionStart: r.secureConnectionStart,
+            serverTiming: r.serverTiming
+                ? (r.serverTiming.map(
+                      (e) => e as PerformanceServerTimingPolyfill
+                  ) as PerformanceServerTimingPolyfill[])
+                : undefined,
```

Comment:
PerformanceServerTiming is unique because it does not have its own entryType for the PerformanceObserver to listen for, but only exists when nested in PerformanceResourceTiming and PerformanceNavigationTiming. Instead, PerformanceServerTimings are manually added by the server to breakdown the requests, sort of like a custom timing API. I'm not sure that we actually want to record this as a separate RUM event, but we can omit for now to avoid blocking. 

For example: https://developer.mozilla.org/en-US/docs/Web/API/PerformanceServerTiming#example

Reply:
Removed server timing from this PR.